### DISCUSSION TITLE
Best way to get start & end position of audio currently visible in the viewport?

### DISCUSSION THREAD

**Discussion by rewbs** [katspaugh/wavesurfer.js#3750] [Q&A]
Hi all, I'm synchronising the scroll & zoom of a WaveSurfer instance with other components on a page. I'm wondering if there's a preferred way to get the offset & duration (or start & end) in seconds of the segment of audio currently visible in the viewport with Wavesurfer 7?

With Wavesurfer 6 I used a combination of `drawer.wrapper.scrollLeft`, `drawer.wrapper.container.clientWidth` and `params.minPxPerSec`. Before I go down that path again I thought I'd check if there's a cleaner way these days. :)

**Comment by rewbs** [comment]
This is what I've landed on so far (I happen to be using wavesurfer-react here):

```
                    onScroll={(ws, visibleStart, visibleEnd) => {
                        setVisibleTime({start: visibleStart, end: visibleEnd});
                        console.log({from: 'scroll', start: visibleStart, end: visibleEnd})
                    }}

                    onZoom={(ws, minPxPerSec) => {
                        const visiblePixels = ws.getWrapper().parentElement?.clientWidth
                        if (!visiblePixels || !isNumber(audioSegmentStart) || !isNumber(audioSegmentEnd)) {
                            return;
                        }
                        const visibleStart =  ws.getScroll() / minPxPerSec;
                        const visibleEnd = (ws.getScroll() + visiblePixels) / minPxPerSec;
                        setVisibleTime({start: visibleStart, end: visibleEnd});
                        console.log({from: 'zoom', start: visibleStart, end: visibleEnd})
                    }}
```

**Reply by katspaugh** [reply]
Yep, that's the intended way to do it :)

**Reply by katspaugh** [reply]
Instead of `ws.getWrapper().parentElement?.clientWidth`, you can also call `ws.getWidth()`.
